packages/pkh-webauthn/src/index.ts

```diff
@@ -0,0 +1,247 @@
+    publicKey: Uint8Array,
+    credential: PublicKeyCredential,
+    did: string
+  }
+  export interface KeySelector {
+    seen?: (credentialId: string, pk: Uint8Array) => void
+    select: (credentialId: string, pk0: Uint8Array, pk1: Uint8Array) => Uint8Array|null
+  }
+  export interface AuthenticatorSession {
+    credentialId?: string // Set if PasskeyProvider was initialized in non-discoverable mode
+    publicKey?: Uint8Array // source for did
+    selectors: KeySelector[]
+  }
+
+  /**
+   * Creates a new session that remembers seen credentials
+   * @param {KeySelector?} selector An object implemeting the KeySelector interface.
+   * @param {string?} credentialId Optionally specify which credential to use for all operations if known ahead of time.
+   */
+  export function createSession (selector?: KeySelector, credentialId?: string): AuthenticatorSession {
+    const selectors: Array<KeySelector> = [new LocalStorageKeySelector()]
+    if (selector) selectors.push(selector)
+    return { credentialId, selectors }
+  }
+
+  export async function getAuthMethod (session: AuthenticatorSession): Promise<AuthMethod> {
+    return async (opts: AuthMethodOpts): Promise<Cacao> => {
+      return createCacao(opts, session)
+    }
+  }
+
+  async function createCacao (opts: AuthMethodOpts, session: AuthenticatorSession): Promise<Cacao> {
+    const now = Date.now()
+    // The public key is not known at pre-sign time.
+    // so we sign a "challenge"-block without Issuer attribute
+    const challenge: Cacao = Object.freeze({
+      h: {
+        t: 'caip122'
+      },
+      p: {
+        domain: globalThis.location.hostname,
+        aud: '' + globalThis.location,
+        iss: '',
+        version: '1',
+        nonce: globalThis.crypto.randomUUID(),
+        resources: [],
+        exp: new Date(now + 7 * 86400000).toISOString(), // 1 week
+        nbf: new Date(now).toISOString(),
+        iat: new Date(now).toISOString(),
+        ...opts // let provided options override defaults
+      }
+    })
+
+    const block = await blockFromCacao(challenge) // await CacaoBlock.fromCacao(challenge) when issue resolved
+    // perform sign
+    const { credential, recovered } = await authenticatorSign(block.cid.bytes)
+
+    // @ts-ignore - credential.response does exist.
+    const { response } = credential
+    const { clientDataJSON, signature } = response
+    const authData = getAuthenticatorData(response)
+    const aad = u8a.toString(assertU8(encode({ authData, clientDataJSON })), 'base64url')
+
+    let pk: any
+    for (const selector of session.selectors) {
+      if (typeof selector.select !== 'function') continue
+      if (pk = await selector.select(credential.id, ...recovered)) break
+    }
+    if (!pk) throw new Error('PublicKeySelectionFailed')
+    if (!recovered.find(c => !u8a.compare(c, pk))) throw new Error('UnrelatedPublickey')
+
+    // Insert iss + aad after signature
+    return {
+      ...challenge,
```

Comment:
```suggestion
      h: challenge.h,
```

Should be sufficient.